packages/server/src/handlers/BaseHandler.ts

```diff
@@ -78,4 +80,53 @@ export class BaseHandler extends EventEmitter {
 
     return decodeURIComponent(match[1])
   }
+
+  getConfiguredMaxSize(req: http.IncomingMessage, id: string) {
+    if (typeof this.options.maxSize === 'function') {
+      return this.options.maxSize(req, id)
+    }
+    return this.options.maxSize ?? 0
+  }
+
+  async getBodyMaxSize(
+    req: http.IncomingMessage,
+    info: Upload,
+    configuredMaxSize?: number
```

Comment:
Yes, I thought about this but we will need the `configuredMaxSize` outside of this function too (in the handler), For this reason instead of calling it twice (since it can be an async operation)